packages/core/src/service.ts

```diff
@@ -159,6 +159,18 @@ export function createServiceBinding<S>(
       name = cls.name.replace(/Provider$/, '');
     }
   }
+  if (!name && isDynamicValueProviderClass(cls)) {
+    // Trim `Provider` from the default service name
+    const templateFn = bindingTemplateFor(cls);
```

Comment:
A question about the logic here:
So if the name exists, the template won't be created for `cls`?

Reply:
The binding template is for intermediate use to extract binding metadata from the template function.